internal/provider/helpers_test.go

```diff
@@ -173,3 +132,52 @@ func Test_stringsEqual(t *testing.T) {
 		})
 	}
 }
+
+func Test_mapKeys(t *testing.T) {
```

Comment:
any particular reason we don't just get rid of the function this is testing and use [map.Keys](https://pkg.go.dev/maps#Keys) from the std lib?